src/lib/intl.jsx

```diff
@@ -1,7 +0,0 @@
-const ReactIntl = require('react-intl');
-
-// Add locale data to react intl for all supported languages
-const localeData = require('scratch-l10n').localeData;
-ReactIntl.addLocaleData(localeData);
```

Comment:
By custom, do we have some non-standard locales? Can you give me an example so I can try it out?

For older browsers, the required `Intl` functionality is polyfilled in `src/lib/intl-polyfill.js`. I have also tested that the non-updated `react-intl` in `scratch-gui` seems to play nicely. It still calls `addLocaleData` in `scratch-gui/src/reducers/locales.js`.

Reply:
`ja-Hira` is an example, so are `ab`, `an`, `ht`, `oc`, and `rap` I believe. Chinese is also a bit complicated, since we use `zh-cn` and `zh-tw` instead of `zh-Hans` and `zh-Hant`.

Reply:
@aoneill01 I found a nice comparison by switching to Aragonés (`an` locale).

this is what's expected - note the date on the comment reading "hace 5 años":
<img width="511" alt="Screen Shot 2023-01-31 at 2 28 14 PM" src="https://user-images.githubusercontent.com/3019167/215863373-dfc7bb73-70f1-4efd-9f45-c0a2029be635.png"> 

this is what I see on my local machine - the date on the comment has defaulted back to English (although the formatting is still off):
<img width="512" alt="Screen Shot 2023-01-31 at 2 27 51 PM" src="https://user-images.githubusercontent.com/3019167/215863544-43a06db9-d1a2-4b00-a6ab-49076f05ec5e.png">

I also get a `Missing locale data for locale: "an" in Intl.NumberFormat. Using default locale: "en" as fallback.` error in the console.

Reply:
@corihudson - I have tried to address these issues in the latest commits.

- Thanks for catching that the auto unit selection was only going up to "days." I considered some different options and ended up writing custom logic to select the appropriate unit of time. Created a `RelativeTime` component to encapsulate this.
- Even in browsers that natively support `Intl`, sometimes locale data is missing for some of our locales. The missing ones vary from browser to browser. Updated the polyfilling logic in `intl-polyfill.js`.
- Some of the locales were not available in the polyfill library. Locales like `ab`, `an`, `ht`, `oc`, and `rap` are swapped out for alternative locales when passed to the `IntlProvider`. Some code locations that were using the locale name from the `IntlProvider` were updated to get the true value instead.

I have compared relative times from these changes to production. For the most part it appears to match. I noted a few differences. I have discussed these with Chris G. None of these were deal breakers for her, and future updates to `Intl` and the polyfill libraries may improve the situation.

- Some relative time translations improved due to better polyfilled locale data, such as Hausa (ha) and کوردیی ناوەندی  (ckb).
- This is likely browser dependent, but I had a couple of locales that were natively supported by my browser. However, the result of the relative time translations was inferior to production. Azeri (az) and Oʻzbekcha (uz) used single-letter placeholders for the time unit.
- Sepedi (nso) and Setswana (tn) fell back to English because there was not polyfill data. Not a big loss since production just shows single-letter placeholders for time units.